go/ngql/types.go

```diff
@@ -419,32 +419,41 @@ func getListElements(vrw types.ValueReadWriter, v types.Value, args map[string]i
 	}
 
 	// Clamp ranges
-	if count <= 0 || idx >= len {
+	if count <= 0 || idx >= end {
 		return ([]interface{})(nil)
 	}
 	if idx < 0 {
 		idx = 0
 	}
-	if idx+count > len {
-		count = len - idx
+	if idx+count > end {
+		count = end - idx
 	}
 
 	values := make([]interface{}, count)
 
-	// This is a workaround for allowing types.Set as types.List.
-	type listOrSetIterator interface {
-		Next() types.Value
-	}
+	// @raf a few questions:
+	//  - why do we do MaybeGetScalar below?
+	//  - in the unittests how could you force the collection to have tree height > 1?
+	//  - where does the rejiggering of the tree height happen?
```

Comment:
Not fully sure what you mean here? The sequence chunker builds and mutates prollytrees... it keeps track of the tree level.